feat: Support authenticating to Azure state backend without a connection string (#8215)
meltano/meltano: docs/docs/reference/settings.mdx

---

### <a name="state-backend-uri"></a>`state_backend.azure.storage_account_url`

- [Environment variable](/guide/configuration#configuring-settings): `MELTANO_STATE_BACKEND_AZURE_STORAGE_ACCOUNT_URL`
- Default: None

One can [Sign-in to Azure](https://learn.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-python?tabs=managed-identity%2Croles-azure-portal%2Csign-in-azure-cli#sign-in-and-connect-your-app-code-to-azure-using-defaultazurecredential) using the Azure CLI.

> The `storage_account_url` which looks like this `https://<myStorageAccountName>.blob.core.windows.net` will be used to feth the default Azure credentals with which your host system is utilizing.

Reviewer: ```suggestion
The `storage_account_url` which looks like `https://<myStorageAccountName>.blob.core.windows.net` and it will be used to fetch the default Azure credentials in the host system.
```

> ### Note
> Leaving `state_backend.azure.storage_account_url` unset will cause Meltano to try to use `state_backend.azure.connection_string`.

Reviewer: ```suggestion
:::info
At least one of `state_backend.azure.storage_account_url` and `state_backend.azure.connection_string` has to be set in order to use Azure Blob Storage as a state backend.

If `state_backend.azure.storage_account_url` is not set, Meltano will to try to use `state_backend.azure.connection_string`. 
:::
```

---

### <a name="state-backend-uri"></a>`state_backend.azure.storage_account_url`

- [Environment variable](/guide/configuration#configuring-settings): `MELTANO_STATE_BACKEND_AZURE_STORAGE_ACCOUNT_URL`
- Default: None

One can [Sign-in to Azure](https://learn.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-python?tabs=managed-identity%2Croles-azure-portal%2Csign-in-azure-cli#sign-in-and-connect-your-app-code-to-azure-using-defaultazurecredential) using the Azure CLI.

The `storage_account_url` which looks like `https://<myStorageAccountName>.blob.core.windows.net` and it will be used to fetch the default Azure credentials in the host system.

:::info

> At least one of `state_backend.azure.storage_account_url` and `state_backend.azure.connection_string` has to be set in order to use Azure Blob Storage as a state backend.
> 
> If `state_backend.azure.storage_account_url` is not set, Meltano will to try to use `state_backend.azure.connection_string`.

Reviewer: ```suggestion
At least one of `state_backend.azure.storage_account_url` and `state_backend.azure.connection_string` has to be set in order to use Azure Blob Storage as a state backend. If `state_backend.azure.storage_account_url` is not set, Meltano will to try to use `state_backend.azure.connection_string`.
```
